app/models/ems_refresh.rb

```diff
@@ -171,7 +171,7 @@ def self.queue_merge(targets, ems, create_task = false)
 
     # Items will be naturally serialized since there is a dedicated worker.
     MiqQueue.put_or_update(queue_options) do |msg, item|
-      targets = msg.nil? ? targets : (msg.args[0] | targets)
+      targets = msg.nil? ? targets : msg.data.concat(targets)
```

Comment:
gives ```undefined method `concat' for nil:NilClass (NoMethodError)```

we need to do
```
(msg.data || []).concat(targets)
```

Reply:
Hm that's strange I didn't get this, what case did you hit that?  That would have to be if the message existed but didn't get any targets assigned?

Reply:
this happens to me everytime that new MiqQueue record is created, the default data seems to be nil

Reply:
If you have a new MiqQueue record, then msg will be nil and you'll hit the first part of the ternary.

Reply:
Migration sounds like the best option :+1: